libnavui-app/src/main/java/com/mapbox/navigation/ui/app/internal/controller/RoutePreviewStateController.kt

```diff
@@ -75,18 +83,22 @@ class RoutePreviewStateController(private val store: Store) : StateController()
             is RoutePreviewAction.Failed -> {
                 RoutePreviewState.Failed(action.reasons, action.routeOptions)
             }
+            else -> state
         }
     }
 
+    // This SAGA is responsible for processing route fetch request.
+    // It uses MapboxNavigation.requestRoutes() and dispatches relevant 'progress' actions when
+    // request starts, cancels, fails or successfully finishes (StartedFetchRequest, Canceled, Failed, Ready)
     private suspend fun MapboxNavigation.fetchRouteSaga() {
         store.actionsFlowable()
             .filter {
-                it is RoutePreviewAction.FetchOptions ||
+                it is RoutePreviewAction.FetchRoute ||
```

Comment:
I know this is already a part of `main`, but I am thinking as to why are we filtering `DestinationAction.SetDestination` if are not doing anything when we `collectLatest` routeOptions which in this case is `null`. Do you think we can remove this?

Reply:
We need this to cancel any in-flight route requests when the destination changes.